src/gdb_main.c

```diff
@@ -148,10 +154,15 @@ int gdb_main_loop(target_controller_s *tc, char *pbuf, size_t pbuf_size, size_t
 	}
 	case 'G': { /* 'G XX': Write general registers */
 		ERROR_IF_NO_TARGET();
-		uint8_t gp_regs[target_regs_size(cur_target)];
-		unhexify(gp_regs, &pbuf[1], sizeof(gp_regs));
-		target_regs_write(cur_target, gp_regs);
-		gdb_putpacketz("OK");
+		const size_t reg_size = target_regs_size(cur_target);
+		if (reg_size) {
+			uint8_t gp_regs[reg_size];
+			unhexify(gp_regs, &pbuf[1], sizeof(gp_regs));
+			target_regs_write(cur_target, gp_regs);
+			gdb_putpacketz("OK");
+		} else {
+			gdb_putpacketz("00");
```

Comment:
This isn't a valid response for 'G' to make per [GDB's docs on the remote protocol](https://docs.jade.fyi/gnu/gdb/gdb.html#index-G-packet); We have two options available here, either returning an error code such as "EFF", or returning "OK" and pretending it worked - we're inclined to the latter, which allows this to be simplified by moving the gdb_putpacketz() below the if block and drop the else.